src/libs/actions/Task.js

```diff
@@ -336,9 +359,25 @@ function reopenTask(taskReportID, taskTitle) {
     ];
 
     // Update optimistic data for parent report action
-    const optimisticParentReportData = ReportUtils.getOptimisticDataForParentReportAction(taskReportID, reopenedTaskReportAction.created, CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD);
-    if (!_.isEmpty(optimisticParentReportData)) {
-        optimisticData.push(optimisticParentReportData);
+    const optimisticDataForParentReportAction = ReportUtils.getOptimisticDataForParentReportAction(taskReportID, reopenedTaskReportAction.created, CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD);
```

Comment:
Coming from https://github.com/Expensify/App/issues/25548
This PR didn't account for the case where a `reportAction` has more than one parent (i.e. if you change the asignee of the task), so only the first one was updated
More details in https://github.com/Expensify/App/issues/25548#issuecomment-1685357377